Sources/SwiftProtobuf/AnyMessageStorage.swift

```diff
@@ -301,163 +323,173 @@ extension AnyMessageStorage {
-  //    orders, so the messages could be identical, but in binary form
-  //    they could differ.
-  public func hash(into hasher: inout Hasher) {
-    if !_typeURL.isEmpty {
-      hasher.combine(_typeURL)
-    }
-  }
-#else  // swift(>=4.2)
-  var hashValue: Int {
-    var hash: Int = i_2166136261
-    if !_typeURL.isEmpty {
-      hash = (hash &* i_16777619) ^ _typeURL.hashValue
-    }
-    return hash
-  }
-#endif  // swift(>=4.2)
-
-  func isEqualTo(other: AnyMessageStorage) -> Bool {
-    if (_typeURL != other._typeURL) {
-      return false
-    }
-
-    // Since the library does lazy Any decode, equality is a very hard problem.
-    // It things exactly match, that's pretty easy, otherwise, one ends up having
-    // to error on saying they aren't equal.
-    //
-    // The best option would be to have Message forms and compare those, as that
-    // removes issues like map<> serialization order, some other protocol buffer
-    // implementation details/bugs around serialized form order, etc.; but that
-    // would also greatly slow down equality tests.
-    //
-    // Do our best to compare what is present have...
-
-    // If both have messages, check if they are the same.
-    if case .message(let myMsg) = state, case .message(let otherMsg) = other.state, type(of: myMsg) == type(of: otherMsg) {
-      // Since the messages are known to be same type, we can claim both equal and
-      // not equal based on the equality comparison.
-      return myMsg.isEqualTo(message: otherMsg)
-    }
-
-    // If both have serialized data, and they exactly match; the messages are equal.
-    // Because there could be map in the message, the fact that the data isn't the
-    // same doesn't always mean the messages aren't equal. Likewise, the binary could
-    // have been created by a library that doesn't order the fields, or the binary was
-    // created using the appending ability in of the binary format.
-    if case .binary(let myValue) = state, case .binary(let otherValue) = other.state, myValue == otherValue {
-      return true
-    }
-
-    // If both have contentJSON, and they exactly match; the messages are equal.
-    // Because there could be map in the message (or the JSON could just be in a different
-    // order), the fact that the JSON isn't the same doesn't always mean the messages
-    // aren't equal.
-    if case .contentJSON(let myJSON, _) = state,
-       case .contentJSON(let otherJSON, _) = other.state,
-       myJSON == otherJSON {
-      return true
-    }
-
-    // Out of options. To do more compares, the states conversions would have to be
-    // done to do comparisions; and since equality can be used somewhat removed from
-    // a developer (if they put protos in a Set, use them as keys to a Dictionary, etc),
-    // the conversion cost might be to high for those uses.  Give up and say they aren't equal.
-    return false
-  }
+      #if swift(>=4.2)
```

Comment:
Certain `#if` blocks don't get formatted at all, if the current version of the compiler just sends us back an `UnknownStmt/Decl`. Note that these are still indented +2. This should improve with newer versions of the compiler, which have more of the syntax nodes properly filled in.

We do the minimum, of ensuring that the left margin of the block is indented to the correct level.

Reply:
One could argue persuasively for indenting everything inside the `#if` block one level further.  I'm agnostic on this personally, but in principle it seems like showing `#if` block structure is just as important as showing brace structure.

Reply:
I agree; a non-scientific and non-exhaustive look at the places where I use `#if` the most is when checking language version or target platform, and they almost always have an `#else` block as well. Indenting those blocks ensures that those two blocks don't blend together into a single harder-to-decipher unit.